specifyweb/frontend/js_src/lib/components/Statistics/Categories.tsx

```diff
@@ -136,6 +137,10 @@ export function Categories({
     (items ?? []).some(
       (item) => item.type === 'CustomStat' || item.isVisible === undefined
     );
+
+  const [deletingCategory, setDeletingCategory] = React.useState<
+    number | undefined
+  >(undefined);
```

Comment:
(Optional) 
Is there a better name for this variable? 
It is not initially clear what type `deletingCategory` is just by looking at the name. 
These may be too long, but something like `[deletingCategoryIndex, setDeletingCategoryIndex]` makes it _very_ clear exactly what the variables reference at a glance. 

However, other developers can simply look at the type, see how it is used, or use IDE tools to know the type. 

I'll leave it up to you which approach you think is better: the more concise current version, or the more verbose but marginally clearer version.

Reply:
could also make it a bit more verbose (but clearer)

```tsx
const [state, setState] = React.useState<State<'Main> | State<'CategoryDelete',{readonly index: number}>>({type:'Main'});
```

OR even better:
move this state into a separate, smaller component - that component will draw the button, the confirmation dialog, and only call handle delete when deletion is confirmed (and handle delete will be passed to the component as a prop)

that solution works because the delete dialog is modal, meaning if one dialog is open, the user can't open another one - thus we don't need a single central state as there is no case when user needs to have two such dialogs open

Reply:
Fixed.